examples/GstHelloWorld.hs

```diff
@@ -0,0 +1,128 @@
+{-# LANGUAGE OverloadedLabels  #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Main where
+
+-- This is a translation of the Hello World gstreamer example to haskell
+-- Original C source: 
+-- https://gstreamer.freedesktop.org/documentation/application-development/basics/helloworld.html
+-- Written by Jaro Reinders on 2017-01-09
+--
+-- Most comments are copied from the source C code 
+-- with the exception of comments surrounded by [ and ]
+
+import qualified GI.GLib as GLib
+import qualified GI.Gst  as Gst
+
+import Data.GI.Base.Properties (setObjectPropertyString)
+
+import Control.Monad (void, when)
+import System.Environment
+import System.IO (stderr)
+import System.Exit (exitFailure, exitSuccess)
+import Data.Text (pack)
+import qualified Data.Text.IO as T
+import Data.Monoid ((<>))
+
+busCall :: GLib.MainLoop -> Gst.Bus -> Gst.Message -> IO Bool
+busCall loop _bus message = do
+  messageTypes <- Gst.getMessageType message
+  
+  when (Gst.MessageTypeEos `elem` messageTypes) $ do
+    putStrLn "End of stream"
+    GLib.mainLoopQuit loop
+
+  when (Gst.MessageTypeError `elem` messageTypes) $ do
+    (gerror,_debug) <- Gst.messageParseError message
+    T.hPutStrLn stderr . ("Error: " <>) =<< Gst.gerrorMessage gerror
+    GLib.mainLoopQuit loop
+  
+  return True
+
+onPadAdded :: Gst.Element -> Gst.Pad -> IO ()
+onPadAdded decoder pad = do
+  -- We can now link this pad with the vorbis-decoder sink pad
+  putStrLn "Dynamic pad created, linking demuxer/decoder"
+  Just sinkPad <- Gst.elementGetStaticPad decoder "sink"
+  void $ Gst.padLink pad sinkPad
+  Gst.objectUnref sinkPad
+
+main :: IO ()
+main = do
+  progName <- getProgName
+  args <- getArgs
+
+  -- Initialization
+  void $ Gst.init Nothing
+
+  loop <- GLib.mainLoopNew Nothing False
+
+  -- Check input arguments
+  when (length args /= 1) $ do
+    T.hPutStrLn stderr $ "Usage: " <> pack progName <> " <Ogg/Vorbis filename>"
+    exitFailure
+  let filename = head args
+
+  -- Crate gstreamer elements
+  pipeline <- Gst.pipelineNew (Just "audio-player")
+  maySource  <- Gst.elementFactoryMake "filesrc"       (Just "file-source")
+  mayDemuxer <- Gst.elementFactoryMake "oggdemux"      (Just "ogg-demuxer")
+  mayDecoder <- Gst.elementFactoryMake "vorbisdec"     (Just "vorbis-decoder")
+  mayConv    <- Gst.elementFactoryMake "audioconvert"  (Just "converter")
+  maySink    <- Gst.elementFactoryMake "autoaudiosink" (Just "audio-output")
+
+  -- [I think this is ugly...]
+  [source,demuxer,decoder,conv,sink] <- maybe
+     (do T.hPutStrLn stderr "One element could not be created. Exiting." 
+         exitFailure 
+         undefined) 
+     return
+     (sequence [maySource, mayDemuxer, mayDecoder, mayConv, maySink])
+
```

Comment:
Maybe instead of this we could define a helper
```haskell
  let makeElement factoryName name =
        Gst.elementFactoryMake factoryName (Just name)
        >>= maybe (error . unpack $ "Could not create " <> name) return
```
and use that?